libs/MeshKernel/include/MeshKernel/TriangulationWrapper.hpp

```diff
@@ -83,117 +73,165 @@ namespace meshkernel
+
+                m_faceNodesFlat.resize(estimatedNumberOfTriangles * 3);
+                std::fill(m_faceNodesFlat.begin(), m_faceNodesFlat.end(), 0);
+
+                m_edgeNodesFlat.resize(estimatedNumberOfTriangles * 2);
+                std::fill(m_edgeNodesFlat.begin(), m_edgeNodesFlat.end(), 0);
+
+                m_faceEdgesFlat.resize(estimatedNumberOfTriangles * 3);
+                std::fill(m_faceEdgesFlat.begin(), m_faceEdgesFlat.end(), 0);
+
+                m_xCoordFlat.resize(estimatedNumberOfTriangles * 3, constants::missing::doubleValue);
+                std::fill(m_xCoordFlat.begin(), m_xCoordFlat.end(), 0.0);
+
+                m_yCoordFlat.resize(estimatedNumberOfTriangles * 3, constants::missing::doubleValue);
+                std::fill(m_yCoordFlat.begin(), m_yCoordFlat.end(), 0.0);
+
                 Triangulation(&intTriangulationOption,
-                              &xLocalPolygon[0],
-                              &yLocalPolygon[0],
+                              xLocalPolygon.data(),
+                              yLocalPolygon.data(),
                               &numInputNodes,
-                              &faceNodesFlat[0], // INDX
-                              &numFaces,
-                              &edgeNodesFlat[0], // EDGEINDX
-                              &numEdges,
-                              &faceEdgesFlat[0], // TRIEDGE
-                              &xNodesFlat[0],
-                              &yNodesFlat[0],
-                              &numNodes,
+                              m_faceNodesFlat.data(), // INDX
+                              &m_numFaces,
+                              m_edgeNodesFlat.data(), // EDGEINDX
+                              &m_numEdges,
+                              m_faceEdgesFlat.data(), // TRIEDGE
+                              m_xCoordFlat.data(),
+                              m_yCoordFlat.data(),
+                              &m_numNodes,
                               &averageTriangleArea);
                 if (estimatedNumberOfTriangles > 0)
                 {
-                    estimatedNumberOfTriangles = -numFaces;
+                    estimatedNumberOfTriangles = -m_numFaces;
                 }
             }
+        }
 
-            m_numFaces = numFaces <= 0 ? static_cast<size_t>(0) : static_cast<size_t>(numFaces);
-            m_numEdges = static_cast<size_t>(numEdges);
-            m_numNodes = static_cast<size_t>(numNodes);
+        /// @brief Build the internal triangulation from the flat triangulation
+        void BuildTriangulation();
 
-            // Create nodes
-            m_nodes.resize(m_numNodes);
-            for (size_t i = 0; i < m_numNodes; ++i)
-            {
-                m_nodes[i] = {xNodesFlat[i], yNodesFlat[i]};
-            }
+        /// @brief Gets the number of triangulated edges
+        /// @return The number of triangulated edges
+        [[nodiscard]] const auto& GetNumEdges() const
```

Comment:
Return by value, it's a just an int.
Personally, I'm not big fan of return type deduction. It makes code harder to read and maintain.